collectors/proc.plugin/proc_pagetypeinfo.c

```diff
@@ -0,0 +1,333 @@
+        ff = procfile_open(config_get(CONFIG_SECTION_PLUGIN_PROC_PAGETYPEINFO, "filename to monitor", ff_path), " \t:", PROCFILE_FLAG_DEFAULT);
+
+        if(unlikely(!ff)) {
+            strncpyz(ff_path, PLUGIN_PROC_MODULE_PAGETYPEINFO_NAME, FILENAME_MAX);
+            ff = procfile_open(PLUGIN_PROC_MODULE_PAGETYPEINFO_NAME, " \t,", PROCFILE_FLAG_DEFAULT);
+        }
+    }
+    if(unlikely(!ff))
+        return 1;
+
+    ff = procfile_readall(ff);
+    if(unlikely(!ff))
+        return 0; // we return 0, so that we will retry to open it next time
+
+    // --------------------------------------------------------------------
+    // Init: find how many Nodes, Zones and Types
+    if(unlikely(pagelines_cnt == 0)) {
+        size_t nodenumlast = -1;
+        char *zonenamelast = NULL;
+
+        ff_lines = procfile_lines(ff);
+        if(unlikely(!ff_lines)) {
+            error("PLUGIN: PROC_PAGETYPEINFO: Cannot read %s, zero lines reported.", ff_path);
+            return 1;
+        }
+
+        // Configuration
+        do_global = config_get_boolean(CONFIG_SECTION_PLUGIN_PROC_PAGETYPEINFO, "enable system summary", CONFIG_BOOLEAN_YES);
+        do_detail = config_get_boolean_ondemand(CONFIG_SECTION_PLUGIN_PROC_PAGETYPEINFO, "enable detail per-type", CONFIG_BOOLEAN_AUTO);
+        filter_types = simple_pattern_create(
+                config_get(CONFIG_SECTION_PLUGIN_PROC_PAGETYPEINFO, "hide charts id matching", "")
+                , NULL
+                , SIMPLE_PATTERN_SUFFIX
+        );
+
+        pagelines_cnt = 0;
+
+        // Pass 1: how many lines would be valid
+        for (l = 4; l < ff_lines; l++) {
+            if (!pagetypeinfo_line_valid(ff, l))
+                continue;
+
+            pagelines_cnt++;
+        }
+        if (pagelines_cnt == 0) {
+            error("PLUGIN: PROC_PAGETYPEINFO: Unable to parse any valid line in %s", ff_path);
+            return 1;
+        }
+
+        // 4th line is the "Free pages count per migrate type at order". Just substract these 8 words.
+        pageorders_cnt = procfile_linewords(ff, 3);
+        if (pageorders_cnt < 9) {
+            error("PLUGIN: PROC_PAGETYPEINFO: Unable to parse Line 4 of %s", ff_path);
+            return 1;
+        }
+
+        pageorders_cnt -= 9;
+
+        if (pageorders_cnt > MAX_PAGETYPE_ORDER) {
+            error("PLUGIN: PROC_PAGETYPEINFO: pageorder found (%lu) is higher than max %d", pageorders_cnt, MAX_PAGETYPE_ORDER);
+            return 1;
+        }
+
+        // Init pagelines from scanned lines
+        if (!pagelines) {
+            pagelines = callocz(pagelines_cnt, sizeof(struct pageline));
+            if (!pagelines) {
+                error("PLUGIN: PROC_PAGETYPEINFO: Cannot allocate %lu pagelines of %lu B", pagelines_cnt, sizeof(struct pageline));
+                return 1;
+            }
+        }
+
+        // Pass 2: Scan the file again, with details
+        p = 0;
```

Comment:
It looks like you don't use it anymore.

Reply:
I do, for `pgl = &pagelines[p];` as I can't use "l" which is the line (starts at 4 & have gaps due to `pagetypeinfo_line_valid`).